### PR TITLE
01 - Use tslib to reduce transpiled files size.

### PR DESCRIPTION
**Pull request by jerome-benoit**
Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

### CODE REVIEW

**Comment by ST-DDT** [comment]
This looks exactly the same as #1. Did you merge the other PR here by accident?

**Comment by jerome-benoit** [comment]
> This looks exactly the same as #1. Did you merge the other PR here by accident?

That PR is stacked over the '00 - ...' because people on GitHub are not used to read changes at the commit level inside a PR, like it's done on most FOSS project using git but not GitHub for the code flow, which then become extremely efficient compared to the ones on GitHub.

**Comment by ST-DDT** [comment]
It might be more efficient for you to stack them on top of each other,
but IMO it makes the PR way harder to review and thus less likely to be merged.
Especially since the PR view doesn't show the actual branches, so the devs have to download the PR locally, identify the relevant changes and then return to the browser to upload the review.
(The currently ongoing update merge commits, don't really help with that either)

I searched for git PR best practices and only found articles that recommend "small changes".
- https://holgerfrohloff.de/best-practices-on-doing-pull-requests
- https://github.com/grpc/grpc-java/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests (Project by Google)

Mind sharing a link that describes your workflow in more details and maybe even does a comparison between both variants?
(I usually merge my PRs in a separate branch (in my own repo) so that I can use all of them without complicating the review itself. This also avoids issues caused by rejected PRs that I still need myself.)

TLDR:

- #1 is functionally independent of #2, but due to the commits interweaving PR #2 depends on #1.
  Thus if PR #1 would be rejected, so would #2.
- From my experience, small PRs are more likely to be merged.

**Comment by jerome-benoit** [comment]
> 
> I searched for git PR best practices and only found articles that recommend "small changes".
> 
> * [holgerfrohloff.de/best-practices-on-doing-pull-requests](https://holgerfrohloff.de/best-practices-on-doing-pull-requests)
> * [https://github.com/grpc/grpc-java/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests](https://github.com/grpc/grpc-java/blob/master/CONTRIBUTING.md?rgh-link-date=2021-02-06T20%3A14%3A12Z#guidelines-for-pull-requests) (Project by Google)

These articles still assume a code flow via web interfaces. 
Keep in mind that git have been designed by Linus Torvalds for a patches (diff generated) based code flow via emails. 
PR were designed to ease the merge of a pile of patches that might depend on each others. 
In a PR on Linux kernel, git, ... projects, you have a series of commits (= patches) that usually contains changes over one topic but also contains cleanups done by the developer. 
They can cover all fixes, features, cleanups, etc. made over, let's say, the network stack, etc.  
The code review is done before such PR merge at the commits level. 
That way, you have an efficient code flow that handle all possible code changes and the code improve a rapid pace. And developper maintain their atomic changes at the commit level and knows how to do it. 

> Mind sharing a link that describes your workflow in more details and maybe even does a comparison between both variants?
> (I usually merge my PRs in a separate branch (in my own repo) so that I can use all of them without complicating the review itself. This also avoids issues caused by rejected PRs that I still need myself.)
> 
> TLDR:
> 
> * #1 is functionally independent of #2, but due to the commits interweaving PR #2 depends on #1.
>   Thus if PR #1 would be rejected, so would #2.
> * From my experience, small PRs are more likely to be merged.

People that use git exclusively via GitHub or similar are doing that. But that's not efficient ;)
You can have a look that kernel.org code flow for examples, you will see that most PRs contain a lot of changes of all kind. The review is done via mailing at the commit (patches) level.
